app/src/components/market/market_home_container.tsx

```diff
@@ -13,25 +14,66 @@ const PAGE_SIZE = 10
 const MarketHomeContainer: React.FC = () => {
   const context = useConnectedWeb3Context()
 
-  const [filter, setFilter] = useState<MarketFilter>(MarketFilter.allMarkets())
-  const [count, setCount] = useState(PAGE_SIZE)
+  const [filter, setFilter] = useState<any>({
+    state: 'OPEN',
+    category: 'All',
+    searchText: '',
+    sortBy: null,
+  })
+  const [markets, setMarkets] = useState<RemoteData<any>>(RemoteData.notAsked())
+  const [cpkAddress, setCpkAddress] = useState<Maybe<string>>(null)
+  const { library: provider } = context
 
-  const { markets, moreMarkets } = useMarkets(context, filter, count)
+  const { data, error, fetchMore, loading } = useQuery(MARKETS_HOME[filter.state], {
+    notifyOnNetworkStatusChange: true,
+    variables: { first: PAGE_SIZE, skip: 0, account: cpkAddress, ...filter },
+  })
 
-  const showMore = () => setCount(count => count + PAGE_SIZE)
-  const onFilterChange = (filter: MarketFilter) => {
-    setCount(PAGE_SIZE)
+  useEffect(() => {
+    const getCpkAddress = async () => {
+      const cpk = await CPKService.create(provider)
+      setCpkAddress(cpk.address)
+    }
+    getCpkAddress()
+  }, [provider])
+
+  useEffect(() => {
+    if (loading) {
+      setMarkets(markets => (RemoteData.hasData(markets) ? RemoteData.reloading(markets.data) : RemoteData.loading()))
+    } else if (error) {
+      setMarkets(RemoteData.failure(error))
+    } else if (data) {
+      if (data.fixedProductMarketMakers.length) {
+        const { fixedProductMarketMakers } = data
+        setMarkets(RemoteData.success(fixedProductMarketMakers))
+      }
+    }
+  }, [data, loading, error])
+
+  const showMore = () => {
+    fetchMore({
+      variables: {
+        skip: data.fixedProductMarketMakers.length,
+      },
+      updateQuery: (prev, { fetchMoreResult }) => {
+        if (!fetchMoreResult) return prev
+        return Object.assign({}, prev, {
+          fixedProductMarketMakers: [...prev.fixedProductMarketMakers, ...fetchMoreResult.fixedProductMarketMakers],
+        })
+      },
+    })
+  }
+  const onFilterChange = (filter: any) => {
```

Comment:
Since this doesn't have extra logic anymore, we can just do `onFilterChange={setFilter}` below.